fission-web-server/library/Fission/Web/Server/Types.hs

```diff
@@ -140,6 +155,106 @@ instance MonadDB (Transaction Server) Server where
 instance MonadRelayStore Server where
   getStoreVar = asks linkRelayStoreVar
 
+instance MonadPowerDNS Server where
+  set recType url zoneTxt contents ttl = do
+    logDebug $ mconcat
+      [ "Set DNS "
+      , displayShow recType
+      , " Record "
+      , displayShow url
+      , " to "
+      , displayShow contents
+      ]
+    clientEnv <- pdnsClientEnv
+    let
+      rrset =
+        RRSet { rrset_name = Text.append (textDisplay url) "."
+              , rrset_type = recType
+              , rrset_ttl = ttl
+              , rrset_changetype = Just Replace
+              , rrset_records = Just $ formatRecords contents
+              , rrset_comments = Nothing
+              }
+      rrsets = RRSets [rrset]
+    resp <- liftIO $ runClientM (PDNS.updateRecords "localhost" zoneTxt rrsets) clientEnv
+    case resp of
+      Left err -> do
+        logWarn @Text "PowerDNS.set failed"
+        return . Left $ Web.Error.toServerError err
+      Right good -> do
+        logDebug @Text "PowerDNS.set succeeded"
+        return $ Right good
+    where
+      formatRecords :: NonEmpty Text -> [Record]
+      formatRecords values =
+        fmap newRecord (toList values)
```

Comment:
👩‍🏫 100% totally fine to write as you have it here. You can use infix backticks or the `<$> = fmap` alias  to break up the line instead of brackets:

```haskell
newRecord `fmap` toList values
newRecord <$> toList values
```

Reply:
The pun here is that `$` is infix application, so `<$>` is mapping (i.e. application to all elements in a container).

Same goes for `&` and `<&>`, which are the same but with arguments reversed